alloydb/notebooks/batch_embeddings_update.ipynb

```diff
@@ -0,0 +1,1089 @@
+        "\n",
+        "df = pd.read_csv(data)\n",
+        "df['etf'] = df['etf'].map({'t': True, 'f': False})\n",
+        "df['rating'] = df['rating'].astype(str).fillna('')"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/",
+          "height": 345
+        },
+        "id": "4R6tzuUtLypO",
+        "outputId": "270d5fcd-b62d-4e3c-8c4e-25428798a350"
+      },
+      "outputs": [],
+      "source": [
+        "df.head()"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "UstTWGJyL7j-"
+      },
+      "source": [
+        "The data consists of the following columns:\n",
+        "\n",
+        "* **id**\n",
+        "* **ticker**: A string representing the stock symbol or ticker (e.g., \"AAPL\" for Apple, \"GOOG\" for Google).\n",
+        "* **etf**: A boolean value indicating whether the asset is an ETF (True) or not (False).\n",
+        "* **market**:  A string representing the stock exchange where the asset is traded.\n",
+        "* **rating**: Whether to hold, buy or sell a stock.\n",
+        "* **overview**: A text field for a general overview or description of the asset.\n",
+        "* **analysis**: A text field, for a more detailed analysis of the asset.\n",
+        "* **overview_embedding** (empty)\n",
+        "* **analysis_embedding** (empty)"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 21,
+      "metadata": {
+        "id": "KqpLkwbWCJaw"
+      },
+      "outputs": [],
+      "source": [
+        "create_table_cmd = sqlalchemy.text(\n",
+        "    f'CREATE TABLE {table_name} ( \\\n",
+        "        id SERIAL PRIMARY KEY, \\\n",
+        "        ticker VARCHAR(255) NOT NULL UNIQUE, \\\n",
+        "        etf BOOLEAN, \\\n",
+        "        market VARCHAR(255), \\\n",
+        "        rating TEXT,  \\\n",
+        "        overview TEXT, \\\n",
+        "        overview_embedding VECTOR (768), \\\n",
+        "        analysis TEXT,  \\\n",
+        "        analysis_embedding VECTOR (768) \\\n",
+        "    )'\n",
+        ")\n",
+        "\n",
+        "\n",
+        "insert_data_cmd = sqlalchemy.text(\n",
+        "    f\"\"\"\n",
+        "    INSERT INTO {table_name} (id, ticker, etf, market,\n",
+        "      rating, overview, analysis) VALUES (:id, :ticker, :etf, :market,\n",
+        "      :rating, :overview, :analysis)\n",
+        "    \"\"\"\n",
+        ")\n",
+        "\n",
+        "parameter_map = [\n",
+        "    {\n",
+        "        \"id\": row[\"id\"],\n",
+        "        \"ticker\": row[\"ticker\"],\n",
+        "        \"etf\": row[\"etf\"],\n",
+        "        \"market\": row[\"market\"],\n",
+        "        \"rating\": row[\"rating\"],\n",
+        "        \"overview\": row[\"overview\"],\n",
+        "        \"analysis\": row[\"analysis\"],\n",
```

Comment:
What's the context on parameter map here? Any reason we can't pass in "row" directly?  Are we trying to reduce the keys in the map?

Reply:
Changed. Used df.to_dict('records') to pass the data instead. This way is cleaner.